packages/sw-runtime-caching/src/lib/request-wrapper.js

```diff
@@ -184,11 +181,11 @@ class RequestWrapper {
       assert.isInstance({request}, Request);
     }
 
-    const clonedRequest = this.pluginCallbacks.fetchDidFail ?
+    const clonedRequest = this.plugins.has('fetchDidFail') ?
       request.clone() : null;
-    if (this.pluginCallbacks.requestWillFetch) {
-      for (let callback of this.pluginCallbacks.requestWillFetch) {
-        const returnedPromise = callback({request});
+    if (this.plugins.has('requestWillFetch')) {
+      for (let plugin of this.plugins.get('requestWillFetch')) {
```

Comment:
Nit: I have a feeling it would be more memory efficient to get the array once and then iterate over the local instance of the array rather than use this short hand.